packages/core/src/store/reducer.ts

```diff
@@ -97,16 +98,31 @@ export function reducer(state: State, action: Action): State {
     }
 
     function save(action: Undoable) {
-      if (draft.history) {
-        draft.history.actions.push([action])
-        draft.history.redoStack = []
-      } else {
+      if (!draft.history) {
         draft.history = {
           initialState: state,
-          actions: [[action]],
+          actions: [],
           redoStack: []
         }
+        debounceTimeout = null
       }
+      if (action.debounce) {
```

Comment:
Would be nice if we could abstract that away and pass as a default config instead. Maybe with a signature similarly to `reduce`, e.g.
```
const commit = (uncommitedActions: Action[], action: Action): [Action[] /* will be commited */, Action[] /* new aggregator */] => {}
```
or something similarly

Reply:
Don't quite understand how that abstraction would work... maybe we can add that when we need it